src/fsharp/IlxGen.fs

```diff
@@ -4781,54 +4781,87 @@ and GenDelegateExpr cenv cgbuf eenvouter expr (TObjExprMethod((TSlotSig(_, deleg
                 false
          with _ ->
             false
+
+    // When creating a delegate to a static method, we can directly reference the method (rather than creating a closure class).
+    let staticMethodFromBody =
+        fun body ->
+            match body with
+            | Expr.App (f, _, _, _, m) ->
+                match f with
```

Comment:
The two top matches can possibly be combined if you feel it remains readable enough, it will save one nesting level.

```fsharp
match body with
| Expr.App (Expr.Val (vref, _, _), _, _, _, m) ->
 ...
```

Reply:
Ah, nice - yes, I remember seeing something similar in the existing code.